### PR TITLE
Add grid of chart and refactor

### PR DESCRIPTION
**Pull request by bobhaslett**
Update bringing this template into line with the common poll tracker
• Add annotation module
• Add popup
• Add grid of charts functionality
• Remove old datasets and change data bindings to reflect
• Update chart render functions to allow control of line, dot and area visibility
• Add auto footnote toggle
• General. clean up

### CODE REVIEW

File: src/core/state.js [hunk 1]
```diff
@@ -35,20 +37,38 @@ export default {
...
-	dotSize: 2.5,
+	tickFormat: "%Y",
 	dotOpacity: 0.2,
-	polls:{smallSize: 1.1, largeSize: 1.25, smallOpacity: 1, largeOpacity: 1},
-	averages:{smallStrokeWidth: 2, largeStrokeWidth: 3, smallOpacity: 1, largeOpacity: 1},
-	moe:{opacityMob: 0.3, opacityDesk: 0.3},
+	averages:{render: true, smallStrokeWidth: 2, largeStrokeWidth: 3, smallOpacity: 1, largeOpacity: 1},
+	moe:{render: true,
+		opacityMob: 0.3,
+		opacityDesk: 0.3},
```
Comment (aendra-rininsland) [inline src/core/state.js:67]: Is this linted?

File: src/parseData.js [hunk 2]
```diff
@@ -53,13 +53,15 @@ export function extentMulti(data, columns) {
   }, {});
   return [ext.min, ext.max];
 }
+
 // A function that returns an array of poll data for a given party
 export function getDots(d, group) {
   const dotsData = [];
-  d.forEach((el) => {
+  d.map((el) => {
```
Comment (aendra-rininsland) [inline src/parseData.js:60]: This is still a forEach, you don't return anything in it.

File: template.yml [hunk 3]
```diff
@@ -37,139 +86,160 @@ settings:
...
   - Popups & panels
   - property: popup
     import: "@flourish/info-popup"
-  # - Legend
-  # - property: legend_container
-  #   import: "@flourish/legend/container"
+  - Footnote controls
+  - property: footnoteSwitch
+    name: Footnote switch
+    description: Generates an autmotic footnote giving tha latest poll in the dataset
```
Comment (aendra-rininsland) [inline template.yml:173]:
```suggestion
    description: Generates an automatic footnote giving the latest poll in the dataset
```

**Review by aendra-rininsland — APPROVED** [review]
This is really good! A few minor style notes but otherwise LGTM — please address the three comments I made then feel free to merge. 👍

File: src/parseData.js (hunk 2 above)
Comment (bobhaslett) [reply src/parseData.js:60]: Done

File: template.yml (hunk 3 above)
Comment (bobhaslett) [reply template.yml:173]: Done

**Pull request merged**